tools/dotcms-cli/cli/src/main/java/com/dotcms/cli/command/DotPush.java

```diff
@@ -55,15 +56,22 @@ default boolean isGlobalPush() {
     WorkspaceManager workspaceManager();
 
     default Optional<Workspace> workspace(){
-       return workspaceManager().findWorkspace(getPushMixin().path());
+        final Path path = getPushMixin().path();
+        final WorkspaceManager workspaceManager = workspaceManager();
+        // This should only really concern Integration tests.
+        // Workspace param is hidden we only use for testing in the Push commands
+        final File workspace = getPushMixin().workspace;
+
+        //If No explicit workspace is provided, we rely on the path to find the workspace
+       return workspaceManager.findWorkspace( workspace != null ? workspace.toPath() : path);
     }
 
     default Path workingRootDir() {
         final Optional<Workspace> workspace = workspace();
         if (workspace.isPresent()) {
             return workspace.get().root();
         }
-        throw new IllegalArgumentException("No valid workspace found.");
+        return getPushMixin().path();
```

Comment:
Do we need this?, it is already calling `workspace()`.

Reply:
This is a fallback if no workspace is found we try the path

Reply:
But just wondering, at that point, after all the logic in the `workspace()` method, if we got to that last line, it does not mean `getPushMixin().path();` is not inside a valid workspace?

Reply:
My thinking here was this: This method is currently consumed from the DotExecutionStategy at this point we don't have any injection points meaning that we don't have Mixins or any of the other cool stuff to report errors.
So If I run this from any folder outside a workspace I'm going to get this exception shown without the right formatting. Because it'll never get handled by the ExcptionHandler 
So If I simply return the path. The exception will still occur..  but it will get reported later from inside the command th getting executed not too early from the DotExceutionStrategy